src/pages/index.js

```diff
@@ -4,92 +4,170 @@ import {
+  selectedCard = null;
+  selectedCardId = null;
+  closeModal(deleteModal);
+});
+
+deleteForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  if (!selectedCardId) return;
+
+  setButtonText(deleteSubmitBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard({ id: selectedCardId })
+    .then(() => {
+      selectedCard.remove();
+      selectedCard = null;
+      selectedCardId = null;
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(deleteSubmitBtn, false, "Delete", "Deleting...");
+    });
+});
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_is-opened");
+    if (openModal) closeModal(openModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_is-opened");
+  document.addEventListener("keydown", handleEscClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+
+modals.forEach((modal) => {
+  modal.addEventListener("click", (evt) => {
+    if (
+      evt.target === modal ||
+      evt.target.classList.contains("modal__close-btn") ||
+      evt.target.classList.contains("modal__close_type_preview")
+    ) {
+      closeModal(modal);
+    }
+  });
+});
+
 function getCardElement(data) {
   const cardElement = cardTemplate.cloneNode(true);
   const cardTitleEl = cardElement.querySelector(".card__title");
   const cardImageEl = cardElement.querySelector(".card__image");
+  const cardDeleteBtnEl = cardElement.querySelector(".card__delete-btn");
+  const cardLikeBtnEl = cardElement.querySelector(".card__like-btn");
 
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
   cardTitleEl.textContent = data.name;
 
-  const cardLikeBtnEl = cardElement.querySelector(".card__like-btn");
+  if (data.isLiked) cardLikeBtnEl.classList.add("card__like-btn_active");
+
   cardLikeBtnEl.addEventListener("click", () => {
-    cardLikeBtnEl.classList.toggle("card__like-btn_active");
+    const isLiked = cardLikeBtnEl.classList.contains("card__like-btn_active");
+    if (!isLiked) {
+      api
+        .addLike(data._id)
+        .then(() => cardLikeBtnEl.classList.add("card__like-btn_active"))
+        .catch(console.error);
+    } else {
+      api
+        .removeLike(data._id)
+        .then(() => cardLikeBtnEl.classList.remove("card__like-btn_active"))
+        .catch(console.error);
+    }
   });
 
-  const cardDeleteBtnEl = cardElement.querySelector(".card__delete-btn");
-  cardDeleteBtnEl.addEventListener("click", () => {
-    cardElement.remove();
-  });
+  cardDeleteBtnEl.addEventListener("click", () =>
+    handleCardDelete(cardElement, data._id)
+  );
 
   cardImageEl.addEventListener("click", () => {
+    const previewImageEl = previewModal.querySelector(".modal__image");
+    const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine